packages/gatsby/src/query/graphql-runner.ts

```diff
@@ -208,6 +201,72 @@ export default class GraphQLRunner {
     // cache just wastes memory without much benefits.
     // TODO: consider a better strategy for cache purging/invalidation
     this.scheduleClearCache()
-    return Promise.resolve(result)
+
+    const result = await promise
+    if (tracer) {
+      tracer.end()
+    }
+    return result
+  }
+}
+
+/**
+ * Tracks and knows how to get a parent span for a particular
+ *  point in query resolver for a particular query and path
+ */
+class GraphQLSpanTracer implements IGraphQLSpanTracer {
+  parentActivity: IPhantomReporter
+  activities: Map<string, IPhantomReporter>
+
+  constructor(name: string, activityArgs: IActivityArgs) {
+    this.parentActivity = report.phantomActivity(name, activityArgs)
+    this.activities = new Map()
```

Comment:
Can you confirm that this is not becoming a `Map<any, any>` without explicit cast here?

Reply:
It's not, but this made me notice a wrong IPhantomActivity import. Thanks!